lib/ruby_indexer/lib/ruby_indexer/collector.rb

```diff
@@ -227,10 +255,15 @@ def add_class_entry(node)
         superclass.slice
       end
 
-      @index << Entry::Class.new(fully_qualify_name(name), @file_path, node.location, comments, parent_class)
+      @index << Entry::Class.new(
+        fully_qualify_name(name),
+        @file_path,
+        node.location,
+        comments,
+        parent_class,
+      )
       @stack << name
-      visit(node.body)
-      @stack.pop
+      @queue.unshift(node.body, LEAVE_EVENT)
```

Comment:
Instead of creating a general purpose event name, I wonder if we should be queueing `Proc` instances, so that the collector can handle `Proc`s and call them.

That way, you will be able to queue any custom behaviour that you want without having to add extra event names.

This line would become:
```suggestion
      @queue.unshift(node.body, -> { @stack.pop })
```

Reply:
If you're okay with it, I would defer that decision until we have a use case for it. Currently, there's no other action we want to take other than popping the stack and I'm not sure about the performance impact of the procs.